xarray/core/datatree_io.py

```diff
@@ -70,27 +79,62 @@ def _datatree_to_netcdf(
     if unlimited_dims is None:
         unlimited_dims = {}
 
-    for node in dt.subtree:
-        at_root = node is dt
-        ds = node.to_dataset(inherit=write_inherited_coords or at_root)
-        group_path = None if at_root else "/" + node.relative_to(dt)
-        ds.to_netcdf(
-            target,
-            group=group_path,
-            mode=mode,
-            encoding=encoding.get(node.path),
-            unlimited_dims=unlimited_dims.get(node.path),
-            engine=engine,
-            format=format,
-            compute=compute,
-            **kwargs,
-        )
-        mode = "a"
+    scheduler = get_dask_scheduler()
+    have_chunks = any(
+        v.chunks is not None for node in dt.subtree for v in node.variables.values()
+    )
+    autoclose = have_chunks and scheduler in ["distributed", "multiprocessing"]
+
+    root_store = get_writable_netcdf_store(
+        target,
+        engine,  # type: ignore[arg-type]
+        mode=mode,
+        format=format,
+        autoclose=autoclose,
+        invalid_netcdf=invalid_netcdf,
+        auto_complex=auto_complex,
+    )
+    if group is not None:
+        root_store = root_store.get_child(group)
```

Comment:
```suggestion
        root_store = root_store.get_child_store(group)
```

:) Are we missing a test?

Reply:
oops, this code path is not curently reachable. I've removed it for now.